packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3060,61 +3056,59 @@ export class ContainerRuntime
 	public orderSequentially<T>(callback: () => T): T {
 		let checkpoint: IBatchCheckpoint | undefined;
 		const checkpointDirtyState = this.dirtyContainer;
-		let result: T;
 		if (this.mc.config.getBoolean("Fluid.ContainerRuntime.EnableRollback")) {
 			// Note: we are not touching any batches other than mainBatch here, for two reasons:
 			// 1. It would not help, as other batches are flushed independently from main batch.
 			// 2. There is no way to undo process of data store creation, blob creation, ID compressor ops, or other things tracked by other batches.
 			checkpoint = this.outbox.getBatchCheckpoints().mainBatch;
 		}
-		try {
-			this._orderSequentiallyCalls++;
-			result = callback();
-		} catch (error) {
-			if (checkpoint) {
-				// This will throw and close the container if rollback fails
-				try {
-					checkpoint.rollback((message: BatchMessage) =>
-						this.rollback(message.contents, message.localOpMetadata),
-					);
-					// reset the dirty state after rollback to what it was before to keep it consistent
-					if (this.dirtyContainer !== checkpointDirtyState) {
-						this.updateDocumentDirtyState(checkpointDirtyState);
+		const result = this.batchRunner.run(() => {
+			try {
+				return callback();
+			} catch (error) {
```

Comment:
I think an optional error handler would be a nice feature on RunCounter.  Would get rid of a nested try-catch here. And would have a similar feel to `Promise.then(onResolve, onReject)`

Reply:
i think it makes it quite a bit harder to read and use when re-implementing the try/catch behavior. it also then opens the door to needing finally, and i'm really like to the keep the run counter simple and single responsiblity.

Reply:
k